include/envoy/access_log/access_log.h

```diff
@@ -62,9 +62,9 @@ class Filter {
    * Evaluate whether an access log should be written based on request and response data.
    * @return TRUE if the log should be written.
    */
-  virtual bool evaluate(const StreamInfo::StreamInfo& info, const Http::HeaderMap& request_headers,
-                        const Http::HeaderMap& response_headers,
-                        const Http::HeaderMap& response_trailers) PURE;
+  virtual bool evaluate(const StreamInfo::StreamInfo& info, const Http::HeaderMap* request_headers,
+                        const Http::HeaderMap* response_headers,
+                        const Http::HeaderMap* response_trailers) PURE;
```

Comment:
What's the motivation for changing this to use pointers? Unless there's some compelling reason, I think keeping the logic for converting nullptr to an empty HeaderMap in one spot (your InstanceImpl class) seems better than having every filter handle nullptr.

Reply:
I think it's really bizarre to take arguments by reference when in most cases the arg isn't even used. It means we're unnecessarily creating new empty HeaderMaps when we don't need them. If we want to group the logic for replacing nullptr with empty headers that can be done in a cleaner way, but it's only even done in a few of the filter impls. As well, we don't yet offer static registration of AccessLog::Filters so this change can be made without affecting devs.

Reply:
I think in most of the place the headers end up being used in the filters we need to dereference the pointers, so I'd rather do it once at the top of the filter stack.